### PR TITLE
feat(stackdriver): support canonical service labels

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR adds support for canonical service labels to the Stackdriver plugin.  In particular, it adds support for `{source|destination}_canonical_service_{name|namespace}` and `{source|destination}_caonical_revision}`.

Signed-off-by: Douglas Reid <douglas-reid@users.noreply.github.com>

### CODE REVIEW

**Comment by bianpengyuan** [comment]
Test situation should be better after https://github.com/istio/proxy/pull/2690 is merged.

File: extensions/stackdriver/metric/record.cc [hunk 1]
```diff
@@ -33,6 +38,34 @@ void record(bool is_outbound, const ::wasm::common::NodeInfo &local_node_info,
       request_info.request_protocol == ::Wasm::Common::kProtocolGRPC
           ? request_info.request_url_path
           : request_info.request_operation;
+
+  const auto &local_labels = local_node_info.labels();
+  const auto &peer_labels = peer_node_info.labels();
+
+  std::string local_canonical_name = local_node_info.workload_name();
```
Comment (bianpengyuan) [inline extensions/stackdriver/metric/record.cc]: How about this?
```
const auto local_name_iter = local_labels.find(kCanonicalNameLabel);
const string& local_canonical_name = local_name_iter == local_labels.end() ? 
       local_node_info.workload_name() : local_name->second;
```

I think this should compile and save an allocation.

Comment (douglas-reid) [reply extensions/stackdriver/metric/record.cc]: done.

**Comment by douglas-reid** [comment]
PTAL.

**Pull request merged**